### PR TITLE
Upgrade to substrate 2.0.0

### PR DESCRIPTION
**Pull request by dvdplm**
Update dependencies and sort out incompatibilities.

### CODE REVIEW

File: src/frame/contracts.rs [hunk 1]
```diff
@@ -116,6 +116,7 @@ pub struct InstantiatedEvent<T: Contracts> {
 }
 
 #[cfg(test)]
+#[cfg(feature = "integration-tests")]
```
Comment (dvdplm) [inline src/frame/contracts.rs:119]: All tests in this module are integration tests so `cargo check --tests` will warn about unused items without this.

File: src/frame/system.rs [hunk 2]
```diff
@@ -118,7 +118,7 @@ pub trait System {
 }
 
 /// Type used to encode the number of references an account has.
-pub type RefCount = u8;
+pub type RefCount = u32;
```
Comment (dvdplm) [inline src/frame/system.rs:121]: @dvc94ch ty! Was scratching my head on this one. :)

File: src/frame/contracts.rs (hunk 1 above)
Comment (ascjones) [reply src/frame/contracts.rs:119]: Means the individual test ones are redundant now, but anyway I'll fix that in #165

**Review by ascjones — APPROVED** [review]
LGTM (test failure just the usual `pure virtual method called`)

**Pull request merged**